scvi/model/base/_base_model.py

```diff
@@ -224,6 +224,39 @@ def _register_manager_for_instance(self, adata_manager: AnnDataManager):
         instance_manager_store = self._per_instance_manager_store[self.id]
         instance_manager_store[adata_id] = adata_manager
 
+    def deregister_manager(self, adata_manager: Optional[AnnDataManager] = None):
+        """Deregisters an :class:`~scvi.data.AnnDataManager` instance with this model instance and class.
+
+        If `adata_manager` is `None`, deregisters all :class:`~scvi.data.AnnDataManager` instances
+        in both the class and instance-specific manager stores, except for the current
+        :class:`~scvi.data.AnnDataManager` instance associated with this model instance.
+        """
+        cls_manager_store = self._setup_adata_manager_store
+        instance_manager_store = self._per_instance_manager_store[self.id]
+
+        if adata_manager is None:
+            instance_managers_to_clear = list(instance_manager_store.keys())
+            cls_managers_to_clear = list(cls_manager_store.keys())
+        else:
+            cls_managers_to_clear = [adata_manager.adata_uuid]
+            instance_managers_to_clear = [adata_manager.adata_uuid]
+
+        for manager_id in cls_managers_to_clear:
```

Comment:
call this adata_id still because manager.id is a separate thing

Reply:
makes sense